### PR TITLE
Part 1 - Metric notifications

### PR DESCRIPTION
**Pull request by OferLevi85**
1. Added metric interface for a dynamic number of attributes / tags
2. Enabled NotificationManager to send notifications through more than one
NotificationService

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/metrics/Metric.java [hunk 1]
```diff
@@ -86,6 +88,15 @@ default void increment(String metric, String requestDomainName, String principal
         increment(metric, requestDomainName);
     }
 
+    /**
+     * Increment the counter for the specified metric with the specified attributes
+     * @param metric Name of the counter
+     * @param attributes Attributes for the metric (tags)
+     */
+    default void increment(String metric, Map<String, String> attributes) {
```
Comment (OferLevi85) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/metrics/Metric.java:96]: Will implement this in server_common in next PR

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationManager.java [hunk 2]
```diff
@@ -30,22 +31,28 @@
...
+    private List<NotificationService> notificationServices = new ArrayList<>();
     private ScheduledExecutorService scheduledExecutor;
     private List<NotificationTask> notificationTasks;
 
     public NotificationManager(List<NotificationTask> notificationTasks) {
         this.notificationTasks = notificationTasks;
-        String notificationServiceFactoryClass = System.getProperty(NOTIFICATION_PROP_SERVICE_FACTORY_CLASS);
-        if (notificationServiceFactoryClass != null) {
-            NotificationServiceFactory notificationServiceFactory;
-            try {
```
Comment (OferLevi85) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationManager.java:42]: Notification Manager can now send notifications to multiple Notification Services.
Users will need to specify the NotificationServiceFactory classes in a comma-separated property.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/MetricNotificationServiceFactory.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    private final static Logger LOG = LoggerFactory.getLogger(MetricNotificationServiceFactory.class);
+
+    @Override
+    public NotificationService create() {
+        return new MetricNotificationService(loadMetricObject());
+    }
+
+    Metric loadMetricObject() {
+
+        String metricFactoryClass = System.getProperty(NOTIFICATION_PROP_METRIC_FACTORY_CLASS,
```
Comment (OferLevi85) [inline libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/impl/MetricNotificationServiceFactory.java:41]: Load a metric object and pass it to MetricNotificationService.
The metric can be a different implementation than the one used by ZTS / ZMS.

File: libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationManager.java [hunk 4]
```diff
@@ -16,6 +16,7 @@
 
 package com.yahoo.athenz.common.server.notification;
 
+import com.mysql.cj.util.StringUtils;
```
[the remark this answers is not in the document]
Comment (OferLevi85) [reply libs/java/server_common/src/main/java/com/yahoo/athenz/common/server/notification/NotificationManager.java:19]: Yeah, I didn't notice it during auto-import, fixing it

**Review by havetisyan — APPROVED** [review]
👍

**Pull request merged**